### PR TITLE
[AMORO-2253] Mixed Format `optimized-sequence` should not contains the skipped partitions

### LINKED ISSUE #2253: [Bug]: optimizing后数据丢失
### What happened?

通过spark3.3.3 往amoro0.5.1版本管理的iceberg1.3.1表里插入数据，总数据量11710条，刚插入完成时，查询表中数据量正常，经过一次MINOR optimizing 再查表中数据，就只有是7271条数据，经确认，数据ID不重复，以下是相关信息：
### 建表语句 ###
CREATE TABLE  jid.dwd.pat_main1 (
pid string COMMENT 'ID',
    ct timestamp COMMENT '创建时间',
    an string,
    pn string,
    ad date ,
    pd date ,
    db_name string,
primary key (pid)
) USING arctic
PARTITIONED BY (db_name,bucket(8,pid))
TBLPROPERTIES (
    'format-version'='2',
    'write.metadata.previous-versions-max' = '5',
    'write.metadata.delete-after-commit.enabled'= 'true',
    'write.upsert.enabled' = 'true',
    'self-optimizing.enabled' = 'true',
    'change.data.ttl.minutes' = '20',
    'snapshot.change.keep.minutes' = '20',
    'snapshot.base.keep.minutes' = '10',
    'table-expire.enabled' = 'true',
    'self-optimizing.max-file-count' = '1000000',
    'clean-orphan-file.min-existing-time-minutes' = '15',
    'self-optimizing.group' = 'amoro_e_flink',
    'clean-orphan-file.enabled' = 'true'
);
531b46a7f0b2d074d2d0320069b5dd9
查询表和对应的chang，结果如下
c1dae2ed9360dff6ca2ac130efacc07
查看对应的hdfs数据，发现base中少了两个分区
c6ce812db36a22a01544c170a8987e5
92fcd304292889f3ab7c445d1eb9492

### Affects Versions

Amoro0.5.1

### What engines are you seeing the problem on?

Spark

### How to reproduce

_No response_

### Relevant log output

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

When the self-optimizing of a Mixed Format Table skips some partitions due to cost limitations of the optimizing process, the calculation of `fromSequence` and `toSequence` still considers all partitions, resulting in a discrepancy between the `optimized-sequence` and the actual optimized partitions.

Close #2253 . 

## Brief change log

- Mixed Format `optimized-sequence` should not contains the skipped partitions

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java [hunk 1]
```diff
@@ -126,10 +128,10 @@ public List<TaskDescriptor> planTasks() {
     evaluators.sort(Comparator.comparing(PartitionEvaluator::getWeight, Comparator.reverseOrder()));
 
     double maxInputSize = MAX_INPUT_FILE_SIZE_PER_THREAD * availableCore;
-    List<PartitionEvaluator> inputPartitions = Lists.newArrayList();
+    actualPartitionPlans = Lists.newArrayList();
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java:131]: This place is a bit strange. Perhaps we can directly return fromSequence and toSequence as the result?

**Review by zhoujinsong — APPROVED** [review]
LGTM.

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java (hunk 1 above)
Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/OptimizingPlanner.java:131]: This requires us to make refactors to the interface, which may not be suitable to complete in this hotfix.

**Comment by wangtaohz** [comment]
This PR has been cherry-picked to 0.5.x.

**Pull request merged**